src/network/networkd-manager.c

```diff
@@ -1446,7 +1450,7 @@ void manager_free(Manager *m) {
 
                 hashmap_remove(m->links, INT_TO_PTR(link->ifindex));
 
-                link_unref(link);
+                link_free(link);
```

Comment:
I don't think we want to this this way. Forcibly freeing an object if it still has some other references remaining is generally bad. (The exception is where there are known circular references between objects. But that mostly applies to objects of the same type in a graph, like the unit graph, and here we have a hierarchy of objects so this should not be needed). The callbacks are all referenced from `manager->rtnl->reply_callbacks` and `manager->rtnl->reply_callbacks_prioq`. We call `sd_netlink_unref(m->rtnl)` below, and this should get rid of all the callbacks. If it doesn't that seems like an error to be fixed there to me.

Reply:
So, should we introduce `sd_netlink_set_destroy_callback()` similar to `sd_bus_slot_set_destroy_callback()`? I will try later.

Reply:
Yes, it sounds like the right solution. I know this is going to be quite a lot of additional code, but I don't think we have much choice if we want to fix this properly.